offloaders/aws-s3/2.5.1/aws-s3.md

```diff
@@ -0,0 +1,551 @@
+
+    ```bash
+    PULSAR_EXTRA_OPTS="${PULSAR_EXTRA_OPTS} ${PULSAR_MEM} ${PULSAR_GC} -Daws.accessKeyId=ABC123456789 -Daws.secretKey=ded7db27a4558e2ea8bbf0bf37ae0e8521618f366c -Dio.netty.leakDetectionLevel=disabled -Dio.netty.recycler.maxCapacity.default=1000 -Dio.netty.recycler.linkCapacity=1024"
+    ```
+
+* Set the access credentials in `~/.aws/credentials`.
+
+    ```conf
+    [default]
+    aws_access_key_id=ABC123456789
+    aws_secret_access_key=ded7db27a4558e2ea8bbf0bf37ae0e8521618f366c
+    ```
+
+* Assume an IAM role.
+
+    This example uses the `DefaultAWSCredentialsProviderChain` for assuming this role.
+
+    The broker must be rebooted for credentials specified in `pulsar_env` to take effect.
+
+    ```conf
+    s3ManagedLedgerOffloadRole=<aws role arn>
+    s3ManagedLedgerOffloadRoleSessionName=pulsar-s3-offload
+    ```
+
+### Size of block read/write
+
+You can configure the size of a request sent to or read from AWS S3 in the configuration file `broker.conf` or `standalone.conf`. 
+
+Configuration|Description|Default value
+|---|---|---
+`s3ManagedLedgerOffloadReadBufferSizeInBytes`|Block size for each individual read when reading back data from AWS S3.|1 MB
+`s3ManagedLedgerOffloadMaxBlockSizeInBytes`|Maximum size of a "part" sent during a multipart upload to GCS. It **cannot** be smaller than 5 MB. |64 MB
+
+## Configure AWS S3 offloader to run automatically
+
+Namespace policy can be configured to offload data automatically once a threshold is reached. The threshold is based on the size of data that a topic has stored on a Pulsar cluster. Once the topic reaches the threshold, an offloading operation is triggered automatically. 
+
+Threshold value|Action
+|---|---
+0|It causes a broker to offload data as soon as possible.
+Negative value|It disables automatic offloading.
+
+Automatic offload runs when a new segment is added to a topic log. If you set the threshold on a namespace, but few messages are being produced to the topic, offload does not work until the current segment is full.
+
+You can configure the threshold size using CLI tools, such as [pulsarctl](https://streamnative.io/docs/v1.0.0/manage-and-monitor/pulsarctl/overview/) or pulsar-admin.
+
+### Example
+
+This example sets the AWS S3 offloader threshold size to 10 MB using pulsar-admin.
+
+```bash
+bin/pulsar-admin namespaces set-offload-threshold --size 10M my-tenant/my-namespace
+```
+
+> #### Tip
+>
+> For more information about the `pulsar-admin namespaces set-offload-threshold options` command, including flags, descriptions, and default values, see [here](http://pulsar.apache.org/tools/pulsar-admin/2.6.0-SNAPSHOT/#-em-set-offload-threshold-em-). 
+
+## Configure AWS S3 offloader to run manually
+
+For individual topics, you can trigger AWS S3 offloader manually using the following methods:
+
+- Use REST endpoint 
+
+- Use CLI tools (such as [pulsarctl](https://streamnative.io/docs/v1.0.0/manage-and-monitor/pulsarctl/overview/) or pulsar-admin). 
```

Comment:
```suggestion
- Use CLI tools (such as [pulsarctl](https://streamnative.io/docs/v1.0.0/manage-and-monitor/pulsarctl/overview/) or pulsar-admin. 
```

Reply:
Here is: Use CLI tools (such as [pulsarctl](https://streamnative.io/docs/v1.0.0/manage-and-monitor/pulsarctl/overview/) or pulsar-admin).